src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -417,10 +424,52 @@ void Terminal::UpdateAppearance(const ICoreAppearance& appearance)
     // before, and shouldn't be now either.
     _scrollOffset = originalOffsetWasZero ? 0 : static_cast<int>(::base::ClampSub(_mutableViewport.Top(), newVisibleTop));
 
+    // Now that we've finished the hard work of resizing the main buffer and
+    // getting the viewport back into the right spot, we need to ALSO resize the
+    // alt buffer, if one exists. Fortunately, this is easy. We don't need to
+    // worry about the viewport and scrollback at all! The alt buffer never has
+    // any scrollback, so we just need to resize it and presto, we're done.
+    if (_inAltBuffer())
+    {
+        _altBuffer->GetCursor().StartDeferDrawing();
+        // we're capturing `this` here because when we exit, we want to EndDefer on the (newly created) active buffer.
+        auto endDefer = wil::scope_exit([this]() noexcept { _altBuffer->GetCursor().EndDeferDrawing(); });
+
+        // First allocate a new text buffer to take the place of the current one.
+        std::unique_ptr<TextBuffer> newTextBuffer;
+        try
+        {
+            // GH#3848 - Stash away the current attributes
+            const auto oldBufferAttributes = _mainBuffer->GetCurrentAttributes();
+            newTextBuffer = std::make_unique<TextBuffer>(bufferSize,
+                                                         TextAttribute{},
+                                                         0, // temporarily set size to 0 so it won't render.
+                                                         true,
+                                                         _mainBuffer->GetRenderer());
+
+            // start defer drawing on the new buffer
+            newTextBuffer->GetCursor().StartDeferDrawing();
+
+            // We don't need any fancy position information. We're just gonna
+            // resize the buffer, it's gonna be in exactly the place it is now.
+            // There's no scrollback to worry about!
+
+            RETURN_IF_FAILED(TextBuffer::Reflow(*_altBuffer.get(),
+                                                *newTextBuffer.get(),
+                                                _GetMutableViewport(),
+                                                std::nullopt));
+
+            // Restore the active text attributes
+            newTextBuffer->SetCurrentAttributes(oldBufferAttributes);
+            _altBuffer.swap(newTextBuffer);
+        }
+        CATCH_RETURN();
+    }
+
     // GH#5029 - make sure to InvalidateAll here, so that we'll paint the entire visible viewport.
     try
     {
-        _buffer->TriggerRedrawAll();
+        _mainBuffer->TriggerRedrawAll();
     }

```

Comment:
Would this not need to use the active buffer now? Otherwise if we're in the alt buffer then this probably wouldn't have any effect.

Reply:
good catch.